cmd/app/options/options.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Jetstack Ltd. See LICENSE for details.
+package options
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apiserver/pkg/util/term"
+	cliflag "k8s.io/component-base/cli/flag"
+
+	"github.com/jetstack/kube-oidc-proxy/pkg/util"
+)
+
+const (
+	AppName = "kube-oidc-proxy"
+)
+
+type Options struct {
+	OIDCAuthentication *OIDCAuthenticationOptions
+	SecureServing      *SecureServingOptions
+	Client             *ClientOptions
+	App                *KubeOIDCProxyOptions
+	Misc               *MiscOptions
+
+	nfs *cliflag.NamedFlagSets
+}
+
+func New() *Options {
+	nfs := new(cliflag.NamedFlagSets)
+
+	// Add flags to command sets
+	return &Options{
+		OIDCAuthentication: NewOIDCAuthenticationOptions(nfs),
+		SecureServing:      NewSecureServingOptions(nfs),
+		Client:             NewClientOptions(nfs),
+		App:                NewKubeOIDCProxyOptions(nfs),
```

Comment:
I would prefer if we could have the KubeOIDCProxy flags first in the help output (as we had it before). I think if we move that line to the top of the struct we will see it first.